src/main/java/radiounit/ManagedRadioUnitRegistry.java

```diff
@@ -0,0 +1,64 @@
+package radiounit;
+
+import java.util.List;
+
+import common.AlarmStatusLevel;
+import common.RatType;
+import common.Vendor;
+
+/**
+ * Radio Unit Registry to keep track of all Radio Units
+ * 
+ * 
+ * @author esiumat
+ *
+ */
+public class ManagedRadioUnitRegistry extends AbstractManagedRadioUnitRegistry {
+
+	public ManagedRadioUnitRegistry() {
+
+	}
+
+	@Override
+	public void addRadioUnit(String ipAddress, String name, Vendor vendor) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public List<ManagedRadioUnit> getByIpAddress(String ipAddress) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public List<ManagedRadioUnit> getByName(String name) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public List<ManagedRadioUnit> getByVendor(Vendor vendor) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public List<ManagedRadioUnit> getByRatType(RatType ratType) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public List<ManagedRadioUnit> getByAlarmStatus(AlarmStatusLevel alarmStatus) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public List<ManagedRadioUnit> getByState(RadioUnitState state) {
+		// TODO Auto-generated method stub
```

Comment:
This comment is not relate to registry, but I wander if you can create a RadioUnitState enum? It's more suitable to be passed around as parameter.

Reply:
Hi, thanks for the comment, and apologies for taking a while to respond. 

Hmm, that's possible. Can you explain your logic a little bit more? I think we had been using the states since we were using the state pattern, and since we already have those state classes, we can reuse them here. 

Are you saying that we could pass Enums as parameters since they'd be easier in terms of memory usage?

Reply:
I am asking because if this method takes a RadioUnitState object, the user of this method(potentially UI, NMS and mediator) will need to instantiate a RadioUnitState object. But the concrete RadioUnitState class(ActivatedState...) constructor need to take a RadioUnit as parameter.
It is unrealistic for UI to hold a RU object and then pass in to RadioUnitState to instantiate a RadioUnitState object. A RadioUnitState enum will be more suitable for UI to use as parameter.
You can still keep the state classes but change its name property to state enum.

Reply:
Hey Yangrui, sorry for not responding for a little while again. 

I think what you said seems to make sense. I don't quite understand the situation fully, but it seems like there is an issue with our current design that cause poor coupling (based on what you said about the UI holding an RU object). 

Perhaps we could call about this sometime next week?

Reply:
I don't think this will be that difficult to address -- we can just add an enum that will identify a RadioUnitState class.

The RadioUnitState class can hold a constant final member with this enum (set by the constructor), and provide a getter for it

Reply:
Let's merge this one, and address the comment in a future submission